modules/storages/app/services/storages/health_service.rb

```diff
@@ -0,0 +1,100 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2024 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages
+  class HealthService
+    def initialize(storage:)
+      @storage = storage
+    end
+
+    def healthy
+      if @storage.health_status == "healthy"
+        @storage.touch(:health_checked_at)
+      else
+        reason = @storage.health_reason
+
+        @storage.update(health_status: "healthy",
+                        health_changed_at: Time.now.utc,
+                        health_checked_at: Time.now.utc,
+                        health_reason: nil)
+
+        admin_users.each do |admin|
+          ::Storages::StoragesMailer.notify_healthy(admin, @storage, reason).deliver_later
+        end
+      end
+    end
+
+    def unhealthy(reason:)
+      time = Time.now.utc
+
+      if @storage.health_status == "unhealthy"
+        if reason_is_same(reason)
+          @storage.update(health_checked_at: time)
+        else
+          @storage.update(health_changed_at: time,
+                          health_checked_at: time,
+                          health_reason: reason)
+
+          notify_unhealthy_admin_users
+        end
+      else
+        @storage.update(health_status: "unhealthy",
+                        health_changed_at: time,
+                        health_checked_at: time,
+                        health_reason: reason)
+
+        notify_unhealthy_admin_users
+      end
+    end
+
+    private
+
+    def notify_unhealthy_admin_users
+      admin_users.each do |admin|
+        ::Storages::StoragesMailer.notify_unhealthy(admin, @storage).deliver_later
+      end
+    end
+
+    def admin_users
+      User.where(admin: true)
+          .where.not(mail: [nil, ""])
+    end
+
+    def schedule_mail_job(storage)
```

Comment:
I do not see where this method is called. Then it is not clear how `::Storages::HealthStatusMailerJob` gets its way to the queue.